tpm2/tpm2_test.go

```diff
@@ -1084,3 +1084,28 @@ func TestQuote(t *testing.T) {
 		t.Errorf("VerifyPKCS1v15 failed: %v", err)
 	}
 }
+
+func TestReadPublicKey(t *testing.T) {
+	rw := openTPM(t)
+	defer rw.Close()
+
+	keyHandle, pubKey1, err := CreatePrimary(rw, HandleOwner, pcrSelection, emptyPassword, defaultPassword, defaultKeyParams)
```

Comment:
This effectively compares pub.Key() output against itself, since CreatePrimary calls it.
The only thing this test proves is that CreatePrimary and ReadPublic commands return the same Public.

Could you rewrite this test using LoadExternal instead?